### ISSUE TITLE
LinkManagement::getChildren() does not include product ID's (and visibility)

### ISSUE THREAD

**Issue by kanduvisla** [magento/magento2#8176]
When I'm using the Service Contract `Magento\ConfigurableProduct\Api\LinkManagementInterface::getChildren($sku)`, I get an array of child products, but those products don't seem to be complete. For instance, `getId()` and `getVisibility()` returns `NULL` on those child products.

### Preconditions

1. Magento 2.1.3

### Steps to reproduce

Get child products using the `LinkManagementInterface` service contract and try to get ID and visibility of the results:

    public function __construct(
        LinkManagementInterface $linkManagement
    ) {
        $this->linkManagement = $linkManagement;
    }

    public function example()
    {
        $childProducts = $this->linkManagement->getChildren('SKU_OF_CONFIGURABLE');
        foreach ($childProducts as $childProduct) {
            var_dump($childProduct->getId());
            var_dump($childProduct->getVisibility());
        }
    }

### Expected result

I should get the ID and Visibility parameter of the child products

### Actual result

I get `NULL` and `NULL`

### Workaround

A possible workaround is to explicitly load the product, but this causes some overhead:

    foreach ($childProducts as $childProduct) {
        // Use ProductRepositoryInterface to explicitly load the product:
        $childProduct = $this->productRepository->get($childProduct->getSku());
        var_dump($childProduct->getId());
        var_dump($childProduct->getVisibility());
    }

**Comment by KrystynaKabannyk** [comment]
Hello @kanduvisla, could you please describe steps to reproduce in more detains?
Thank you!

**Comment by kanduvisla** [comment]
@KrystynaKabannyk I think my steps are pretty self-explanatory. Just create an instance of `Magento\ConfigurableProduct\Api\LinkManagementInterface` and call the `getChildren()`-method. But if you insist:

Step 1: Create a configurable product with some simples.
Step 2: Create a class with at least this structure:

```php
class LinkManagementBugExample
{
    public function __construct(
        LinkManagementInterface $linkManagement
    ) {
        $this->linkManagement = $linkManagement;
    }

    public function example(string $sku)
    {
        $childProducts = $this->linkManagement->getChildren($sku);
        foreach ($childProducts as $childProduct) {
            var_dump($childProduct->getId());
            var_dump($childProduct->getVisibility());
        }
    }
}
```

Step 3: Run the `example()`-method with the SKU of the configurable.

The resulted output is 2 times `NULL`, rather than the ID and the visibility of the child product.

**Comment by kanduvisla** [comment]
Any updates on this issue? I ran into it again today. As a workaround I have to include the product resource as a dependency:

    $this->productResource->getIdBySku($child->getSku())

**Comment by kanduvisla** [comment]
Still no update on this? Can at least anyone confirm this issue? Or is it just me?

**Comment by magento-engcom-team** [comment]
@kanduvisla, thank you for your report.
We've created internal ticket(s) MAGETWO-81151 to track progress on the issue.

**Comment by okorshenko** [comment]
Hi @kanduvisla 
The issue has been fixed and delivered in 2.2-develop branch. Will be available with upcoming patch release.
